aiohttp_pydantic/view.py

```diff
@@ -24,28 +24,42 @@ class PydanticView(AbstractView):
     An AIOHTTP View that validate request using function annotations.
     """
 
+    # Allowed HTTP methods; overridden when subclassed.
+    allowed_methods: Set[str] = {}
+
     async def _iter(self) -> StreamResponse:
-        method = getattr(self, self.request.method.lower(), None)
-        resp = await method()
-        return resp
+        if (method_name := self.request.method) not in self.allowed_methods:
+            self._raise_allowed_methods()
+        return await getattr(self, method_name.lower())()
 
     def __await__(self) -> Generator[Any, None, StreamResponse]:
         return self._iter().__await__()
 
-    def __init_subclass__(cls, **kwargs):
+    def __init_subclass__(cls, **kwargs) -> None:
+        """Define allowed methods and decorate handlers.
```

Comment:
I think the more easier way to prevents to decorate a handler multiple times is to update the `inject_params` decorator to "tag" the handler.

```
def inject_params(
    handler, parse_func_signature: Callable[[Callable], Iterable[AbstractInjector]]
):
    if  getattr(handler, 'is_aiohttp_pydantic_handler', False):
        return
    ...
    wrapped_handler.is_aiohttp_pydantic_handler = True
    return wrapped_handler
```